src/components/Map.jsx

```diff
@@ -15,6 +15,41 @@ import LayerSwitcher from "./LayerSwitcher";
 import { AbsoluteCenter, Code, Box } from "@chakra-ui/react";
 import { HiOutlinePlusSmall } from "react-icons/hi2";
 
+const addTMSLayer = (map, layerId, url, visibility = true) => {
+  // Check if the layer already exists (avoid duplicates)
+  if (map.getSource(layerId))  {
+    // set the visibility of the layer
+    map.setLayoutProperty(`tms-layer-${layerId}`, "visibility", visibility ? "visible" : "none");
+    return;
+  }
+
+  // Find the first GeoJSON layer ID
+  const layers = map.getStyle().layers;
+  const firstGeoJSONLayer = layers.find((layer) =>
+    layer.id.startsWith("fill-layer-")
+  );
+
+  const beforeId = firstGeoJSONLayer ? firstGeoJSONLayer.id : undefined;
+
+  // Add the TMS source
+  map.addSource(layerId, {
+    type: "raster",
+    tiles: [url],
+    tileSize: 256,
+  });
+
+  // Add the TMS layer with the calculated beforeId
+  map.addLayer(
+    {
+      id: `tms-layer-${layerId}`,
+      type: "raster",
+      source: layerId,
+      layout: { visibility: visibility ? "visible" : "none" },
+    },
+    beforeId // Insert below the first GeoJSON layer
+  );
+};
```

Comment:
Is there a reason that you're not using the ReactMapGL components to add an order layers on the map. Using the components `Source` and `Layer` instead of manually managing this in an effect hook will improve the readability of this component.